nmdc_server/ingest/pipeline.py

```diff
@@ -26,29 +26,25 @@ def __call__(self, db: Session, obj: Dict[str, Any], **kwargs: Any) -> LoadObjec
 
 # Load metagenome annotation as well as the gene function annotations produced.
 def load_mg_annotation(db: Session, obj: Dict[str, Any], **kwargs) -> LoadObjectReturn:
+    logger = get_logger(__name__)
     pipeline = schemas.MetagenomeAnnotationBase(**obj)
     row = models.MetagenomeAnnotation(**pipeline.dict())
     db.add(row)
     db.flush()
 
     annotations: Collection = kwargs["annotations"]
 
-    query = annotations.find(
-        {
-            "was_generated_by": pipeline.id,
-            "has_function": {
-                "$regex": ko_regex,
-            },
-        },
-        no_cursor_timeout=True,
-    )
+    query = annotations.find({"was_generated_by": pipeline.id})
     if kwargs.get("function_limit"):
         query = query.limit(kwargs["function_limit"])
 
     gene_functions: Dict[str, models.GeneFunction] = {}
     mga_gene_functions: List[Tuple[UUID, str, str, str]] = []
     for gf in query:
         function_id = gf["has_function"]
+        if not re.match(ko_regex, function_id):
+            logger.warn(f"Unexpected function {function_id} for pipeline {pipeline.id}")
```

Comment:
I think it's a good practice for warnings to end up in Sentry. If we don't leave the function check out of the query, let's instead keep a running total of how many are being excluded, and only issue a log warning if it's above a certain threshold. That seems like something we'd want to get an alert about.

Reply:
We don't have sentry in celery, this is currently a few hundred messages, and I think the individual values are valuable for knowing what's wrong.  Can we make a second pass for better handling and reporting later?

Reply:
Fine by me. I'll file a ticket; we should definitely have sentry setup for celery.